pysph/base/particle_array.pyx

```diff
@@ -511,7 +507,7 @@ cdef class ParticleArray:
             the type of particles that need to be removed.
 
         """
-        if self.gpu is not None:
+        if self.gpu is not None and self.backend:
```

Comment:
Why do you need this `self.backend`?  Can it be false?  Do you mean `self.backend == 'opencl'`?

Reply:
This is so that you can use the methods defined in ParticleArray even when `self.gpu` is set. For example see `test_resize_works` in `test_device_helper`

Reply:
Maybe I can have a different value of backend for which this happens, like 'cpu' or something?